### DISCUSSION TITLE
Alternative to the now disabled `brew audit [path]` command?

### DISCUSSION THREAD

**Discussion by Justintime50** [Homebrew/discussions#4864] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.17-13-gfa56f72
ORIGIN: https://github.com/Homebrew/brew
HEAD: fa56f723b1b7531c0cd7b100f8cac5fdb7806f4e
Last commit: 7 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 31905ed383d51193b759a51be35341197c06c285
Core tap last commit: 4 weeks ago
Core tap branch: master
Core tap JSON: 25 Oct 21:20 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.42.0 => /opt/homebrew/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4.1-arm64
CLT: N/A
Xcode: 14.3.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  dotnet
  node@14
  openssl@1.1
  php@8.0
  php@7.3
  php@7.4
  ufraw
```

### Description of issue

I maintain a GitHub Action to [release Homebrew formula to a tap](https://github.com/Justintime50/homebrew-releaser). On CI, I would run `brew audit --formula /path/to/test/formulas/*.rb` which would audit all my output formula that were generated via this action. Recently I started receiving the following error: `Error: Calling brew audit [path ...] is disabled! Use brew audit [name ...] instead.`.

I'm curious if there is an alternative to running `brew audit` on a collection of formula locally without needing to define their names. I won't have their names upfront and have a dozen or more that need auditing. I will only have local formula Ruby files to work with. The documentation on the subject is scarce or outdated since this [change was made 3 months ago](https://github.com/Homebrew/brew/commit/7da934f7e203a90608273910760a8569a413fee6).

Hopefully there is an option out there for people with my use case! Deprecating and then removing the ability to do something without also providing an alternative or suggestion in the error or docs is pretty unfortunate.

**Comment by SMillerDev** [comment, marked as answer]
The name should be the same as the file name so you would already have the name. You just need to make sure the files are in a tap for this command to work.

Additionally, this change was made because path based audits never worked correctly so expect some new issues to pop up.

**Reply by Justintime50** [reply]
Thanks for the reply. Since these are autogenerated files with dummy data, they aren't technically real formula. The audit is to ensure we're following convention etc so I wouldn't be able to install them. Could I create a "local" tap, add them all to that tap, audit the tap, without making it live?

**Reply by SMillerDev** [reply]
Yeah, that was exactly what I was suggesting.

**Reply by Justintime50** [reply]
For future travelers, this is how I solved the issue:

```bash
    brew tap-new homebrew-releaser/test --no-git
    cp -r test/formulas/* $(brew --repository)/Library/Taps/homebrew-releaser/homebrew-test/Formula
    cp -r test/formula_imports $(brew --repository)/Library/Taps/homebrew-releaser/homebrew-test
    for file in $(brew --repository)/Library/Taps/homebrew-releaser/homebrew-test/Formula/*
    do
        brew audit --formula "$(basename ${file%.rb})"
    done
    brew untap homebrew-releaser/test
```

Unfortunate my one liner had to break out into this beast 😔 thanks again for pointing me in the right direction!

**Reply by muescha** [reply]
To ensure that you don't overlap with existing names, I would recommend including the tap repository name in the `brew audit` command as follows: `user/repo/name`:

```shell
brew audit --formula "homebrew-releaser/test/$(basename ${file%.rb})"
```

**Reply by ReenigneArcher** [reply]
>     brew tap-new homebrew-releaser/test --no-git

Thank you! I'm creating a similar github action to yours, and this example helped me.

One thing to take note of is it creates the tap under a slightly different name `homebrew-releaser/homebrew-test` instead of what was supplied as `homebrew-releaser/test`.
